simtools/simtel/simtel_histograms.py

```diff
@@ -67,50 +69,184 @@ def get_histogram_title(self, i_hist):
             self.combine_histogram_files()
         return self.combined_hists[i_hist]["title"]
 
-    def combine_histogram_files(self):
-        """Add the values of the same type of histogram from the various lists into a single
-        histogram list."""
-        # Processing and combining histograms from multiple files
-        self.combined_hists = []
-        n_files = 0
+    @property
+    def list_of_histograms(self):
+        """
+        Returns a list with the histograms for each file.
+
+        Returns
+        -------
+        list:
+            List of histograms.
+        """
+
+        self._list_of_histograms = []
         for file in self._histogram_files:
-            count_file = True
             with EventIOFile(file) as f:
                 for o in yield_toplevel_of_type(f, Histograms):
                     try:
                         hists = o.parse()
                     except Exception:  # pylint: disable=broad-except
                         self._logger.warning(f"Problematic file {file}")
-                        count_file = False
                         continue
+                    self._list_of_histograms.append(hists)
+        return self._list_of_histograms
+
+    def _check_consistency(self, first_hist_file, second_hist_file):
+        """
+        Checks whether two histograms have the same format.
+        Raises an error in case they are not consistent.
+
+        Parameters
+        ----------
+        first_hist_file: dict
+            One histogram from a single file.
+        second_hist_file: dict
+            One histogram from a single file.
+
+        Raises
+        ------
+        BadHistogramFormat:
+            if the format of the histograms have inconsistent dimensions.
+        """
+        for key_to_test in [
+            "lower_x",
+            "upper_x",
+            "n_bins_x",
+            "title",
+        ]:
+            if first_hist_file[key_to_test] != second_hist_file[key_to_test]:
+                msg = "Trying to add histograms with inconsistent dimensions"
+                self._logger.error(msg)
+                raise BadHistogramFormat(msg)
```

Comment:
Tiny comment on the naming and you are free to ignore "BadHistogramFormat" indicates that something is wrong with the format of a histogram, while here you are testing for inconsistent dimensions. Maybe `InconsistentHistogramFormat`?

Reply:
Ok, I changed the name of the raise statement as suggested.